mne_bids/utils.py

```diff
@@ -26,9 +27,144 @@
+    space : str | None
+        The coordinate space for an anatomical file. Corresponds to "space".
+    prefix : str | None
+        The prefix for the filename to be created. E.g., a path to the folder
+        in which you wish to create a file with this name.
+    suffix : str | None
+        The suffix for the filename to be created. E.g., 'audio.wav'.
+
+    Examples
+    --------
+    >>> bids_basename = make_bids_basename(subject='test', session='two', task='mytask', suffix='data.csv')  # noqa: E501
+    >>> print(bids_basename)
+    sub-test_ses-two_task-mytask_data.csv
+    >>> print(bids_basename.__repr__())
+    BIDSPath (sub-test_ses-two_task-mytask_data.csv)
+    >>> # copy and update multiple entities at once
+    >>> new_basename = bids_basename.copy().update(subject='test2', session='one')
+    >>> print(new_basename)
+    sub-test2_ses-one_task-mytask_data.csv
+    """
+
+    def __init__(self, subject=None, session=None,
+                 task=None, acquisition=None, run=None, processing=None,
+                 recording=None, space=None, prefix=None, suffix=None):
+        self.update(subject=subject, session=session, task=task,
+                    acquisition=acquisition, run=run, processing=processing,
+                    recording=recording, space=space, prefix=prefix,
+                    suffix=suffix)
+
+    @property
+    def entities(self):
+        """Return dictionary of the BIDS entities."""
+        # create an ordered dictionary of all the bids entities
+        entities = OrderedDict()
+        for key in BIDS_ENTITIES:
+            value = getattr(self, key, None)
+            if value is not None:
+                entities[key] = value
+        return entities
+
+    def __str__(self):
+        """Return the string representation of the path."""
+        basename = _gen_bids_basename(**self.entities)
+        return basename
+
+    def __repr__(self):
+        """Representation in the style of `pathlib.Path`."""
+        return "{}({!r})".format(self.__class__.__name__, str(self))
+
+    def __fspath__(self):
+        """Return the string representation for any fs functions."""
+        return str(self)
+
+    def __bytes__(self):
+        """Return the bytes representation of the path.
+
+        This is only recommended to use under Unix.
+        """
+        return os.fsencode(str(self))
+
+    def __eq__(self, other):
+        """Compare str representations."""
+        return str(self) == str(other)
+
+    def __ne__(self, other):
+        """Compare str representations."""
+        return str(self) != str(other)
+
+    def copy(self):
+        """Copy the instance.
+
+        Returns
+        -------
+        bidspath : instance of BIDSPath
+            The copied bidspath.
+        """
+        return deepcopy(self)
+
+    def update(self, **kwargs):
+        """Update inplace BIDS entity key/value pairs in object.
+
+        Parameters
+        ----------
+        kwargs :
```

Comment:
dtype is missing

Reply:
This is a stupid question... but these are of dtype 'dict' right? lol

Reply:
Also, renamed as `entities`